### PR TITLE
feat(adapter-node-http): Upgrade nock to v11.x

### LINKED ISSUE #263: TypeError: rawRequest.connection.address is not a function
## Description

We are using a library that uses a [Popsicle](https://github.com/serviejs/popsicle) client.

There is a failure in mocking the requests using Popsicle.

### Shareable Source

Pretty vanilla Polly setup:

```ts
import path from 'path'
import { Polly, PollyServer } from '@pollyjs/core'
import NodeHttpAdapter from '@pollyjs/adapter-node-http'
import FSPersister from '@pollyjs/persister-fs'

Polly.register(FSPersister)
Polly.register(NodeHttpAdapter)

export class PollyJS {
  recordingName: string
  polly: Polly

  constructor(recordingName: string) {
    this.recordingName = recordingName
    this.polly = new Polly(this.recordingName, {
      adapters: ['node-http'],
      persister: 'fs',
      persisterOptions: {
        fs: {
          recordingsDir: path.resolve(__dirname, '../__recordings__'),
        },
      },
    })
  }

  getPollyInstance(): Polly {
    return this.polly
  }

  getPollyServer(): PollyServer {
    return this.polly.server
  }
}
```

### Error Message & Stack Trace

```
    TypeError: rawRequest.connection.address is not a function

      at OverriddenClientRequest.onResponse (node_modules/popsicle-transport-http/src/index.ts:316:33)
      at _respond (node_modules/nock/lib/request_overrider.js:525:17)
      at respond (node_modules/nock/lib/request_overrider.js:508:11)
```

The code can be found here: https://github.com/serviejs/popsicle-transport-http/blob/ef78022e431532ec710c716ca91b7d3691afe888/src/index.ts#L316

### Config

See above.

### Dependencies

Copy the @pollyjs dependencies from `package.json`:

```json
{
    "@pollyjs/adapter-node-http": "2.6.3",
    "@pollyjs/core": "2.6.3",
    "@pollyjs/persister-fs": "2.6.3"
}
```

## Relevant Links

Provided by @nguyentoanit:
https://github.com/nguyentoanit/popsicle-polly-error

## Environment

Tell us which operating system you are using, as well as which versions of Node.js and npm/yarn. If applicable, include the browser and the corresponding version.

Run the following to get it quickly:

```
Node.js v10.15.3
darwin 18.7.0
npm 6.11.3
```

### PR DESCRIPTION
**Pull request by offirgolan**
## Motivation and Context

Resolves #263, #264

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] I have added tests to cover my changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] My code follows the code style of this project.
- [x] My commits and the title of this PR follow the [Conventional Commits Specification](https://www.conventionalcommits.org).
- [x] I have read the [contributing guidelines](https://github.com/Netflix/pollyjs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: packages/@pollyjs/adapter-node-http/rollup.config.test.js [hunk 1]
```diff
@@ -1,7 +1,17 @@
 import createNodeTestConfig from '../../../scripts/rollup/node.test.config';
 import createJestTestConfig from '../../../scripts/rollup/jest.test.config';
 
-const external = ['http', 'https', 'url', 'stream', 'crypto'];
+const external = [
```
Comment (jasonmit) [inline packages/@pollyjs/adapter-node-http/rollup.config.test.js]: Could these be exported and then imported in the rollup config?  This way we're only maintaining a single set of external deps

File: packages/@pollyjs/adapter-node-http/src/index.js [hunk 2]
```diff
@@ -63,16 +65,28 @@ export default class HttpAdapter extends Adapter {
...
+            if (!isUtf8Representable(body)) {
+              body = Buffer.from(body, 'hex');
+            }
+
+            body = body.toString('utf8');
+          } else if (typeof body === 'object') {
+            try {
+              body = JSON.stringify(body);
+            } catch (e) {
+              // not a valid JSON string
```
Comment (jasonmit) [inline packages/@pollyjs/adapter-node-http/src/index.js]: Should this swallow?

File: packages/@pollyjs/adapter-node-http/src/index.js [hunk 3]
```diff
@@ -63,16 +65,28 @@ export default class HttpAdapter extends Adapter {
...
-
-        // body will always be a string unless the content-type is application/json
-        // in which nock will then parse into an object. We have our own way of
-        // dealing with json content to convert it back to a string.
-        if (body && typeof body !== 'string') {
-          body = JSON.stringify(body);
+        const contentType = (headers['content-type'] || '').toString();
+        const isMultiPart = contentType.includes('multipart');
+
+        if (body) {
```
Comment (jasonmit) [inline packages/@pollyjs/adapter-node-http/src/index.js]: Seems some overlap between these changes and https://github.com/Netflix/pollyjs/pull/257/files

Comment (offirgolan) [reply packages/@pollyjs/adapter-node-http/src/index.js]: Yeah there is some overlap. @swashcap can validate and we can add some more test cases in that PR.

**Pull request merged**